daemon/containerd/cache.go

```diff
@@ -48,6 +47,24 @@ func (ic *imageCache) GetCache(parentID string, cfg *container.Config) (imageID
 		}
 	}
 
+	if parent == nil { // "FROM scratch"
+		imgs, err := ic.c.client.ImageService().List(ctx) // TODO is this right??
+		if err != nil {
+			return "", err
+		}
+		for _, img := range imgs {
+			id := img.Target.Digest.String()
+			image, err := ic.c.GetImage(ctx, id, imagetype.GetImageOpts{})
+			if err != nil {
+				return "", err
+			}
+			if isMatch(image, parent, cfg) {
+				return id, nil
+			}
+		}
+		return "", nil
+	}
+
 	children, err := ic.c.Children(ctx, parent.ID())
```

Comment:
The new `if` block above is very similar to the one found in `ic.c.Children`, and honestly this _could_ be implemented as passing an empty ID to `Children`, but I haven't done so to compare whether that implementation is simpler yet.